contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+        return _proposalVotes[proposalId].overrideVoteReceipt[account].hasVoted;
+    }
+
+    /**
+     * @dev Accessor to the internal vote counts.
+     */
+    function proposalVotes(
+        uint256 proposalId
+    ) public view virtual returns (uint256 againstVotes, uint256 forVotes, uint256 abstainVotes) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+        return (proposalVote.againstVotes, proposalVote.forVotes, proposalVote.abstainVotes);
+    }
+
+    /**
+     * @dev See {Governor-_quorumReached}.
+     */
+    function _quorumReached(uint256 proposalId) internal view virtual override returns (bool) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return quorum(proposalSnapshot(proposalId)) <= proposalVote.forVotes + proposalVote.abstainVotes;
+    }
+
+    /**
+     * @dev See {Governor-_voteSucceeded}. In this module, the forVotes must be strictly over the againstVotes.
+     */
+    function _voteSucceeded(uint256 proposalId) internal view virtual override returns (bool) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return proposalVote.forVotes > proposalVote.againstVotes;
+    }
+
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        uint256 totalWeight,
+        bytes memory params
+    ) internal virtual override returns (uint256) {
+        if (keccak256(params) == keccak256(hex"23b70c8d0000000000000000000000000000000000000000")) {
+            return _countVotesOverride(proposalId, account, support, params);
+        }
+
+        totalWeight -= _overrideVoteWeight[account][proposalId];
+
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        if (proposalVote.voteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVote(account);
+        }
+        proposalVote.voteReceipt[account] = VoteReceipt({hasVoted: true, support: support});
+
+        if (support == uint8(VoteType.Against)) {
+            proposalVote.againstVotes += totalWeight;
+        } else if (support == uint8(VoteType.For)) {
+            proposalVote.forVotes += totalWeight;
+        } else if (support == uint8(VoteType.Abstain)) {
+            proposalVote.abstainVotes += totalWeight;
+        } else {
+            revert GovernorInvalidVoteType();
+        }
+
+        return totalWeight;
+    }
+
+    function _countVotesOverride(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        bytes memory params
+    ) private returns (uint256) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+        uint256 proposalSnapshot = proposalSnapshot(proposalId);
+        address delegate = _getPastDelegate(account, proposalSnapshot);
+
+        if (proposalVote.overrideVoteReceipt[account].hasVoted) {
+            revert GovernorAlreadyCastVoteOverride(account);
+        }
```

Comment:
Not being able top override before the delegate has voted is a limitation:
- if a delegate vote at the last minute, you don't have time to override
- if a delegate doesn't vote, the token holder can't do anything (meaning the delegate can prevent users to use their votes by not voting at all)

Can we find a way so that is possible?

Reply:
This is a check to ensure that the delegatee does not override vote twice. An override vote can come at any time when voting is allowed.